### PR TITLE
NAS-116162 / 22.12 / NAS-116162: Refactoring VM device forms

### PR DESCRIPTION
**Pull request by undsoft**
Testing: Go to Virtual Machines, create a VM, click Devices. Test Add and Edit forms.

Additional changes:
- Ix-combobox will now show initial loading indicator.

### CODE REVIEW

File: src/app/pages/vm/devices/device-form/device-form.component.ts [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+  typeControl = new FormControl(VmDeviceType.Cdrom, Validators.required);
+  orderControl = new FormControl(null as number);
+
+  cdromForm = this.formBuilder.group({
+    path: ['/mnt', Validators.required],
+  });
+
+  diskForm = this.formBuilder.group({
+    path: [''],
```
Comment (AlexKarpov98) [inline src/app/pages/vm/devices/device-form/device-form.component.ts:56]: path: ['', Validators.required],

File: src/app/pages/vm/devices/device-form/device-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+    type: [null as VmDiskMode],
+    sectorsize: [null as number],
```
Comment (AlexKarpov98) [inline src/app/pages/vm/devices/device-form/device-form.component.ts:58]: sectorsize: [0],

File: src/app/pages/vm/devices/device-form/device-form.component.ts [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+    type: [null as VmDiskMode],
+    size: [null as number],
+  });
+
+  pciForm = this.formBuilder.group({
+    pptdev: ['', Validators.required],
+  });
+
+  displayForm = this.formBuilder.group({
+    port: [null as number],
```
Comment (AlexKarpov98) [inline src/app/pages/vm/devices/device-form/device-form.component.ts:79]: port: [null as number, Validators.required],

File: src/app/pages/vm/devices/device-form/device-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+    resolution: [''],
+    bind: [''],
+    password: ['', Validators.maxLength(8)],
+    type: [VmDisplayType.Vnc],
+    web: [true],
+  });
```
Comment (AlexKarpov98) [inline src/app/pages/vm/devices/device-form/device-form.component.ts:85]: Seems missing `wait` checkbox field.

File: src/app/pages/vm/devices/device-form/device-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+    mac: ['', regexValidator(this.networkService.macRegex)],
+    nic_attach: ['', Validators.required],
+  });
+
+  rawFileForm = this.formBuilder.group({
+    path: ['', Validators.required],
+    sectorsize: [0],
+    type: [null as VmDiskMode],
+    size: [null as number],
+  });
```
Comment (AlexKarpov98) [inline src/app/pages/vm/devices/device-form/device-form.component.ts:72]: Seems missing `rootpwd` & `boot` fields

**Review by AlexKarpov98 — APPROVED** [review]
Screenshot_26
Screenshot_27
Screenshot_28
Screenshot_29

LGTM! 👍  Seems missing some fields.

**Comment by undsoft** [comment]
Yeah, I should have written that some of the fields were old and were removed on purpose.

**Pull request merged**